packages/web3-react/src/components/ConnectModal/ConnectWithWalletConnect.tsx

```diff
@@ -29,16 +29,19 @@ const ConnectWithWalletConnect: React.FC = () => {
   const { connect } = useConnect({
     chainGroup: context.addressGroup,
     keyType: context.keyType,
-    networkId: context.network ?? ''
+    networkId: context.network
   })
 
   useEffect(() => {
     if (_init) {
       return
     }
 
-    _init = true
-    connect().catch((err) => setError(`${err}`))
+    connect()
+      .then(() => {
+        _init = true
```

Comment:
What happens if `error` already has a value before `_init` is set to true? I think we should add the following line inside this function, otherwise, in this particular corner-case, the text below will never show the correct value

```ts
setError(undefined)
```

Reply:
Let's leave this comment to this issue: https://github.com/alephium/alephium-web3/issues/154
I feel that this implementation would have to be rewritten soon, so let's spend too much time for now.

Reply:
Great catch, thanks!

Reply:
I thought it's gonna be a big change due to the global variable `_init` and local react state (which might be tricky)